src/main/java/org/testeditor/fixture/web/WebDriverFixture.java

```diff
@@ -145,10 +154,89 @@ public WebDriver startBrowser(String browser) {
 		}
 		configureDriver();
 		return driver;
-    }
+	}
+
+	public void initWithReporter(TestRunReporter reporter) {
+		reporter.addListener(this);
+	}
+
+	@Override
+	public void reported(SemanticUnit unit, Action action, String msg) {
+		if (unit == SemanticUnit.TEST && action == Action.ENTER) {
+			runningTest = msg;
+		}
+		if (screenshotShouldBeMade(unit, action, msg)) {
+			screenshot(msg + '.' + action.name());
+		}
+	}
+
+	private String runningTest = null;
+	private static final int SCREENSHOT_FILENAME_MAXLEN = 128;
```

Comment:
Windows 7 has a limitation of a file path length of 252 characters incl. file extension.
So a file name like:
/screenshots/testCaseName_with_128_char_182225.123-escapedBaseName_with128_char.png
could lead to  unpredictable results like, files can not be backed up on other directories, or even worse just could not be opened.